### PR TITLE
Make Greater Than or Equal Date Criteria Explicit

### PR DESCRIPTION
**Pull request by mjriley**
## Product Description
Referenced ticket: https://dimagi.atlassian.net/browse/SAAS-15372
This should fix the issue called out in the ticket, where the selected option for the user should now be correctly labelled as 'greater than or equal'. We have added new features if the user wants 'greater than' or 'less than or equal'

image

## Technical Summary
We kept the constants the same for these options so users that previously had 'greater than' comparisons should now see 'greater than or equal'

### Safety story
Added tests and verified locally

### Automated test coverage

New tests added in _corehq.apps.data_interfaces.tests.test_models_

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/data_interfaces/models.py [hunk 1]
```diff
@@ -559,11 +560,11 @@ def matches(self, case, now):
...
 
 class MatchPropertyDefinition(CaseRuleCriteriaDefinition):
-    # True when today < (the date in property_name + property_value days)
-    MATCH_DAYS_BEFORE = 'DAYS_BEFORE'
+    MATCH_DAYS_LESS_THAN = 'DAYS_BEFORE'
+    MATCH_DAYS_LESS_OR_EQUAL = 'DAYS_LESS_OR_EQUAL'
 
-    # True when today >= (the date in property_name + property_value days)
-    MATCH_DAYS_AFTER = 'DAYS'
+    MATCH_DAYS_GREATER_THAN = 'GREATER_THAN_DAYS'
```
Comment (millerdev) [inline corehq/apps/data_interfaces/models.py:566]: The others all start with `DAYS`
```suggestion
    MATCH_DAYS_GREATER_THAN = 'DAYS_GREATER_THAN'
```

**Review by millerdev — COMMENTED** [review]
The test failure looks relevant: `ERROR: corehq.apps.data_interfaces.tests.test_auto_case_updates:CaseRuleCriteriaTest.test_date_case_property_after`

File: corehq/apps/data_interfaces/models.py (hunk 1 above)
Comment (mjriley) [reply corehq/apps/data_interfaces/models.py:566]: Should be fixed now. `_try_date_conversion` apparently returns `datetime` objects when a string is provided, which, unfortunately for us, are also considered instances of `date`. I noticed the two other usages of `_try_date_conversion` in this file have the same logic of forcing the returned `datetime` into a `date` object. I was tempted to just modify the implementation of `_try_date_conversion` to return a `date`, but it looks like there is one other usage, imported by _scheduling/scheduling_partitioned/models_ that specifically wants a `datetime` returned here.  Given that `try_date_conversion` seems to imply that its returning a date anyway, do you think I should modify that function and then just force the `date` into a `datetime` for the _scheduling_partitioned_ code?

**Comment by mjriley** [comment]
> The test failure looks relevant: `ERROR: corehq.apps.data_interfaces.tests.test_auto_case_updates:CaseRuleCriteriaTest.test_date_case_property_after`

Should be fixed now. _try_date_conversion apparently returns datetime objects when a string is provided, which, unfortunately for us, are also considered instances of date. I noticed the two other usages of _try_date_conversion in this file have the same logic of forcing the returned datetime into a date object. I was tempted to just modify the implementation of _try_date_conversion to return a date, but it looks like there is one other usage, imported by scheduling/scheduling_partitioned/models that specifically wants a datetime returned here. Given that try_date_conversion seems to imply that its returning a date anyway, do you think I should modify that function and then just force the date into a datetime for the scheduling_partitioned code?

**Review by millerdev — APPROVED** [review]
> Given that try_date_conversion seems to imply that its returning a date anyway, do you think I should modify that function and then just force the date into a datetime for the scheduling_partitioned code?

Seems reasonable.

**Pull request merged**